### PR TITLE
Source Hubspot: remove API Key authentication type

### LINKED ISSUE #18484: Source Hubspot: review authentication methods due to upcoming API Key sunset

### PR DESCRIPTION
**Pull request by roman-yermilov-gl**
## What
*Hubspot is going to sunset its API Key authentication method, so we need to remove it from the connector*
- updated spec
- fixed code
- fixed tests

*This changes will affect those users who using API Key authentication and `Quotes` stream is no longer available*

### CODE REVIEW

**Review by davydov-d — APPROVED** [review]

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-hubspot/acceptance-test-config.yml [hunk 1]
```diff
@@ -16,16 +16,9 @@ tests:
   discovery:
     - config_path: "secrets/config.json"
       backward_compatibility_tests_config:
-        disable_for_version: "0.1.83"
+        disable_for_version: "0.2.2"
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-hubspot/acceptance-test-config.yml:19]: please include a comment indicating why it was disabled for this version, to give the next developer an understanding of why it was done

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/spec.yaml [hunk 2]
```diff
@@ -65,26 +65,6 @@ connectionSpecification:
...
-            api_key:
-              title: API key
-              description: >-
-                HubSpot API Key. See the <a
-                href="https://knowledge.hubspot.com/integrations/how-do-i-get-my-hubspot-api-key">Hubspot docs</a>
-                if you need help finding this key.
-              type: string
-              airbyte_secret: true
         - type: object
           title: Private APP
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/spec.yaml:69]:
```suggestion
          title: Private App
```

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-hubspot/acceptance-test-config.yml (hunk 1 above)
Comment (girarda) [reply airbyte-integrations/connectors/source-hubspot/acceptance-test-config.yml:19]: should this be a major upgrade if it's not backward compatible?

**Review by girarda — COMMENTED** [review]
how are we going to migrate cloud customers using the api key?

File: airbyte-integrations/connectors/source-hubspot/acceptance-test-config.yml (hunk 1 above)
Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-hubspot/acceptance-test-config.yml:19]: I thought it should. Could you please clarify it?

**Comment by roman-yermilov-gl** [comment]
> how are we going to migrate cloud customers using the api key?

Those customers should switch to one of the other two types of authentication

**Comment by YowanR** [comment]
@girarda Is your concern that this change will be breaking customers' sync?

**Comment by girarda** [comment]
@YowanR yes, customers will need to migrate off the api_key before we can deploy this to cloud because their syncs will break

**Comment by YowanR** [comment]
We need a plan to land this and we should get TCS involved (@erica-airbyte We will need your help here).
Here's what I suggest:
1. Let's identify how many customers will be impacted and then plan to reach out to them to let them know that this change is coming (This is a Hubspot change, not an Airbyte one). As a side note, we should also reach out to all Hubspot customers as an FYI (but that's an AI for me).
2. Once the comms are out, we can push this change in and then merge.

What do folks think?

**Review by lazebnyi — APPROVED** [review]

**Comment by TBernstein4** [comment]
@YowanR @girarda Sorry I did not respond to that last message. I just finished sending messages to all Active Cloud Workspaces with Hubspot Connectors that are using API keys. Maybe wait until Monday Nov. 28 before this is deployed to give users as much time as possible to read the message and make the changes?

**Comment by YowanR** [comment]
Thanks @TBernstein4! I've also posted in [#public-airbyte-help on Slack](https://airbytehq-team.slack.com/archives/C021JANJ6TY/p1668466115922499?thread_ts=1668460093.346059&cid=C021JANJ6TY) so we should be covered on both the OSS and Cloud side. Agree that we should wait until Nov 28th to deploy this.

**Comment by lazebnyi** [comment]
@TBernstein4 it seems we can deploy this PR, isn't it?

**Comment by TBernstein4** [comment]
@lazebnyi Yes, all good to go! Thanks for noticing and asking!

**Pull request merged**